ui/cryptomaterial/modal.go

```diff
@@ -31,19 +31,21 @@ type Modal struct {
 	scroll         ListStyle
 	padding        unit.Dp
 
-	isFloatTitle  bool
-	isDisabled    bool
-	showScrollBar bool
-	isMobileView  bool
+	isFloatTitle         bool
+	isDisabled           bool
+	showScrollBar        bool
+	isMobileView         bool
+	isFirstDisplay       bool
+	firstLoadWithContext func(gtx C)
 }
 
-func (t *Theme) ModalFloatTitle(id string, isMobileView bool) *Modal {
-	mod := t.Modal(id, isMobileView)
+func (t *Theme) ModalFloatTitle(id string, isMobileView bool, firstLoad ...func(gtx C)) *Modal {
```

Comment:
You don't need a variadic parameter here because it requires always checking if it was passed. Just pass a function with an empty body if you don't want to execute anything.

kindly also add a comment on why the extra parameter is necessary.
```suggestion

func (t *Theme) ModalFloatTitle(id string, isMobileView bool, firstLoad func(gtx C)) *Modal {
```

Reply:
Your approach is good too, but it is a option parameter, it is reason it should is  variadic parameter, it should be handled in one place instead of passing functions with empty bodies pointlessly if not needed

Reply:
You can also just pass nil I think if there's only one function and you don't need it.

Reply:
oh, thank you @dmigwi @JoeGruffins